pkgs/tools/compression/zchunk/default.nix

```diff
@@ -0,0 +1,42 @@
+{ stdenv, fetchFromGitHub, fetchpatch, meson, ninja, pkgconfig, curl, openssl, zstd, argp-standalone }:
+
+stdenv.mkDerivation rec {
+  pname = "zchunk";
+  version = "1.1.5";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "13sqjslk634mkklnmzdlzk9l9rc6g6migig5rln3irdnjrxvjf69";
+  };
+
+  patches = [
+    (fetchpatch {
```

Comment:
can you add comments as to why the patches are needed, and when they can be removed

Reply:
Good call, sorry for not getting to this. See below.